companion.py

```diff
@@ -425,45 +476,69 @@ def dump(self, r):
-        for thing in ['buyPrice', 'sellPrice', 'demand', 'demandBracket', 'stock', 'stockBracket']:
+        for thing in ('buyPrice', 'sellPrice', 'demand', 'demandBracket', 'stock', 'stockBracket'):
             if not isinstance(commodity.get(thing), numbers.Number):
-                if __debug__: print('Invalid "%s":"%s" (%s) for "%s"' % (thing, commodity.get(thing), type(commodity.get(thing)), commodity.get('name', '')))
+                if __debug__:
+                    print(
+                        'Invalid {!r}:{!r} ({}) for {!r}'.format(
+                            thing,
+                            commodity.get(thing),
+                            type(commodity.get(thing)),
+                            commodity.get('name', '')
+                        )
+                    )
                 break
+
         else:
-            if not category_map.get(commodity['categoryname'], True):	# Check not marketable i.e. Limpets
+            # Check not marketable i.e. Limpets
+            if not category_map.get(commodity['categoryname'], True):
                 pass
-            elif commodity['demandBracket'] == 0 and commodity['stockBracket'] == 0:	# Check not normally stocked e.g. Salvage
+
+            # Check not normally stocked e.g. Salvage
+            elif commodity['demandBracket'] == 0 and commodity['stockBracket'] == 0:
                 pass
-            elif commodity.get('legality'):	# Check not prohibited
+            elif commodity.get('legality'):  # Check not prohibited
                 pass
+
             elif not commodity.get('categoryname'):
-                if __debug__: print('Missing "categoryname" for "%s"' % commodity.get('name', ''))
+                if __debug__:
+                    print('Missing "categoryname" for {!r}'.format(commodity.get('name', '')))
+
             elif not commodity.get('name'):
-                if __debug__: print('Missing "name" for a commodity in "%s"' % commodity.get('categoryname', ''))
+                if __debug__:
+                    print('Missing "name" for a commodity in {!r}'.format(commodity.get('categoryname', '')))
+
             elif not commodity['demandBracket'] in range(4):
-                if __debug__: print('Invalid "demandBracket":"%s" for "%s"' % (commodity['demandBracket'], commodity['name']))
+                if __debug__:
+                    print('Invalid "demandBracket":{!r} for {!r}'.format(commodity['demandBracket'], commodity['name']))
+
             elif not commodity['stockBracket'] in range(4):
-                if __debug__: print('Invalid "stockBracket":"%s" for "%s"' % (commodity['stockBracket'], commodity['name']))
+                if __debug__:
+                    print('Invalid "stockBracket":{!r} for {!r}'.format(commodity['stockBracket'], commodity['name']))
+
             else:
                 # Rewrite text fields
-                new = dict(commodity)	# shallow copy
+                new = dict(commodity)  # shallow copy
```

Comment:
`dict.copy()` does a shallow copy already, perhaps use that instead of this, for clarity.